cider/tests/nextgen/jitlib/JITLibTest.cpp

```diff
@@ -143,6 +143,176 @@ TEST_F(JITLibTests, LogicalOpTest) {
       false, true, true, [](JITValue& a, JITValue& b) { return !a; });
 }
 
+TEST_F(JITLibTests, BasicIFControlFlowWithoutElseTest) {
+  LLVMJITModule module("TestModule");
+  JITFunctionPointer function = module.createJITFunction(JITFunctionDescriptor{
+      .function_name = "test_func",
+      .ret_type = JITFunctionParam{.type = JITTypeTag::INT32},
+      .params_type = {JITFunctionParam{.name = "x", .type = JITTypeTag::INT32},
+                      JITFunctionParam{.name = "condition", .type = JITTypeTag::BOOL}}});
+  {
+    JITValuePointer ret = function->createVariable("ret", JITTypeTag::INT32);
+    ret = function->getArgument(0);
+    auto if_builder = function->createIfBuilder();
+    if_builder
+        ->condition([&]() {
+          auto condition = function->getArgument(1);
+          return condition;
+        })
+        ->ifTrue([&]() { ret = ret + 1; })
+        ->build();
+    function->createReturn(ret);
+  }
+  function->finish();
+  module.finish();
+
+  auto func_ptr = function->getFunctionPointer<int32_t, int32_t, bool>();
+  EXPECT_EQ(func_ptr(123, false), 123);
+  EXPECT_EQ(func_ptr(123, true), 124);
+}
+
+TEST_F(JITLibTests, BasicIFControlFlowWithElseTest) {
+  LLVMJITModule module("TestModule");
+  JITFunctionPointer function = module.createJITFunction(JITFunctionDescriptor{
+      .function_name = "test_func",
+      .ret_type = JITFunctionParam{.type = JITTypeTag::INT32},
+      .params_type = {JITFunctionParam{.name = "x", .type = JITTypeTag::INT32},
+                      JITFunctionParam{.name = "condition", .type = JITTypeTag::BOOL}}});
+  {
+    JITValuePointer ret = function->createVariable("ret", JITTypeTag::INT32);
+    ret = function->getArgument(0);
+    auto if_builder = function->createIfBuilder();
+    if_builder
+        ->condition([&]() {
+          auto condition = function->getArgument(1);
+          return condition;
+        })
+        ->ifTrue([&]() { ret = ret + 1; })
+        ->ifFalse([&]() { ret = ret + 10; })
+        ->build();
+    function->createReturn(ret);
+  }
+  function->finish();
+  module.finish();
+
+  auto func_ptr = function->getFunctionPointer<int32_t, int32_t, bool>();
+  EXPECT_EQ(func_ptr(123, false), 133);
+  EXPECT_EQ(func_ptr(123, true), 124);
+}
+
+TEST_F(JITLibTests, NestedIFControlFlowTest) {
```

Comment:
What about multi `IFControlFlow` exist in single function?

Reply:
Just create them sequentially.
```c++
auto if_builder1 = func->createIfBuilder();
auto if_builder2 = func->createIfBuilder();
...

if_builder1->condition()->...->build();
if_builder2->condition()->...->build();
...
```